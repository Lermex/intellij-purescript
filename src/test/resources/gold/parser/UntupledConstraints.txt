Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSImportModuleNameImpl(ImportModuleName)
          PSQualifiedImpl(Qualified)
            PsiElement(proper name)('Prelude')
          PsiElement(whitespace)('\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSImportModuleNameImpl(ImportModuleName)
          PSQualifiedImpl(Qualified)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
            PsiElement(proper name)('Console')
          PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
          PSValueRefImpl(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSTypeClassDeclarationImpl(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSQualifiedImpl(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Show')
              PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PSGenericIdentifierImpl(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSImpliesImpl(Implies)
          PsiElement(<=)('<=')
          PsiElement(whitespace)(' ')
        PSClassNameImpl(ClassName)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Nonsense')
            PsiElement(whitespace)(' ')
        PSGenericIdentifierImpl(GenericIdentifier)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSTypeDeclarationImpl(TypeDeclaration)
          PSTypeAnnotationNameImpl(TypeAnnotationName)
            PsiElement(identifier)('method')
            PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSTypeImpl(Type)
            PSTypeAtomImpl(TypeAtom)
              PSGenericIdentifierImpl(GenericIdentifier)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSTypeImpl(Type)
              PSTypeAtomImpl(TypeAtom)
                PSGenericIdentifierImpl(GenericIdentifier)
                  PsiElement(identifier)('a')
                  PsiElement(whitespace)('\n\n')
      PSDataDeclarationImpl(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Box')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          PSGenericIdentifierImpl(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Box')
            PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PSGenericIdentifierImpl(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)('\n\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSTypeInstanceDeclarationImpl(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PSGenericIdentifierImpl(GenericIdentifier)
          PsiElement(identifier)('showBox')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSQualifiedImpl(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Show')
              PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PSGenericIdentifierImpl(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(=>)('=>')
        PsiElement(whitespace)(' ')
        PSClassNameImpl(ClassName)
          PSQualifiedImpl(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Show')
              PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PsiElement(()('(')
          PSTypeImpl(Type)
            PSTypeAtomImpl(TypeAtom)
              PSTypeConstructorImpl(TypeConstructor)
                PSQualifiedImpl(Qualified)
                  PSProperNameImpl(ProperName)
                    PsiElement(proper name)('Box')
                    PsiElement(whitespace)(' ')
            PSTypeAtomImpl(TypeAtom)
              PSGenericIdentifierImpl(GenericIdentifier)
                PsiElement(identifier)('a')
          PsiElement())(')')
          PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSValueDeclarationImpl(ValueDeclaration)
          PsiElement(identifier)('show')
          PsiElement(whitespace)(' ')
          PSNamedBinderImpl(NamedBinder)
            <empty list>
          PSBinderImpl(Binder)
            PsiElement(()('(')
            PSBinderImpl(Binder)
              PSBinderAtomImpl(BinderAtom)
                PSConstructorBinderImpl(ConstructorBinder)
                  PSGenericIdentifierImpl(GenericIdentifier)
                    PSQualifiedImpl(Qualified)
                      PSProperNameImpl(ProperName)
                        PsiElement(proper name)('Box')
                        PsiElement(whitespace)(' ')
                  PSBinderImpl(Binder)
                    PSVarBinderImpl(VarBinder)
                      PsiElement(identifier)('a')
            PsiElement())(')')
            PsiElement(whitespace)(' ')
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSValueImpl(Value)
            PSPrefixValueImpl(PrefixValue)
              PSStringLiteralImpl(StringLiteral)
                PsiElement(string)('"Box "')
                PsiElement(whitespace)(' ')
            PSIdentInfixImpl(IdentInfix)
              PSQualifiedImpl(Qualified)
                PsiElement(operator)('<>')
                PsiElement(whitespace)(' ')
            PSValueImpl(Value)
              PSPrefixValueImpl(PrefixValue)
                PSVarImpl(Var)
                  PSQualifiedImpl(Qualified)
                    PsiElement(identifier)('show')
                    PsiElement(whitespace)(' ')
                PSVarImpl(Var)
                  PSQualifiedImpl(Qualified)
                    PsiElement(identifier)('a')
                    PsiElement(whitespace)('\n\n')
      PSTypeDeclarationImpl(TypeDeclaration)
        PSTypeAnnotationNameImpl(TypeAnnotationName)
          PsiElement(identifier)('strangeThing')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSTypeImpl(Type)
          PSTypeAtomImpl(TypeAtom)
            PSForAllImpl(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PSGenericIdentifierImpl(GenericIdentifier)
                PsiElement(identifier)('m')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSConstrainedTypeImpl(ConstrainedType)
                PSTypeImpl(Type)
                  PSTypeAtomImpl(TypeAtom)
                    PSTypeConstructorImpl(TypeConstructor)
                      PSQualifiedImpl(Qualified)
                        PSProperNameImpl(ProperName)
                          PsiElement(proper name)('Semigroup')
                          PsiElement(whitespace)(' ')
                  PSTypeAtomImpl(TypeAtom)
                    PsiElement(()('(')
                    PSTypeImpl(Type)
                      PSTypeAtomImpl(TypeAtom)
                        PSGenericIdentifierImpl(GenericIdentifier)
                          PsiElement(identifier)('m')
                          PsiElement(whitespace)(' ')
                      PSTypeAtomImpl(TypeAtom)
                        PSTypeConstructorImpl(TypeConstructor)
                          PSQualifiedImpl(Qualified)
                            PSProperNameImpl(ProperName)
                              PsiElement(proper name)('Unit')
                    PsiElement())(')')
                    PsiElement(whitespace)(' ')
                  PsiElement(=>)('=>')
                  PsiElement(whitespace)(' ')
                  PSTypeImpl(Type)
                    PSTypeAtomImpl(TypeAtom)
                      PSGenericIdentifierImpl(GenericIdentifier)
                        PsiElement(identifier)('m')
                        PsiElement(whitespace)(' ')
                    PSTypeAtomImpl(TypeAtom)
                      PSTypeConstructorImpl(TypeConstructor)
                        PSQualifiedImpl(Qualified)
                          PSProperNameImpl(ProperName)
                            PsiElement(proper name)('Unit')
                            PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSTypeImpl(Type)
                      PSTypeAtomImpl(TypeAtom)
                        PSGenericIdentifierImpl(GenericIdentifier)
                          PsiElement(identifier)('m')
                          PsiElement(whitespace)(' ')
                      PSTypeAtomImpl(TypeAtom)
                        PSTypeConstructorImpl(TypeConstructor)
                          PSQualifiedImpl(Qualified)
                            PSProperNameImpl(ProperName)
                              PsiElement(proper name)('Unit')
                              PsiElement(whitespace)(' ')
                      PsiElement(->)('->')
                      PsiElement(whitespace)(' ')
                      PSTypeImpl(Type)
                        PSTypeAtomImpl(TypeAtom)
                          PSGenericIdentifierImpl(GenericIdentifier)
                            PsiElement(identifier)('m')
                            PsiElement(whitespace)(' ')
                        PSTypeAtomImpl(TypeAtom)
                          PSTypeConstructorImpl(TypeConstructor)
                            PSQualifiedImpl(Qualified)
                              PSProperNameImpl(ProperName)
                                PsiElement(proper name)('Unit')
                                PsiElement(whitespace)('\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PsiElement(identifier)('strangeThing')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('x')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('y')
        PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSVarImpl(Var)
              PSQualifiedImpl(Qualified)
                PsiElement(identifier)('x')
                PsiElement(whitespace)(' ')
          PSIdentInfixImpl(IdentInfix)
            PSQualifiedImpl(Qualified)
              PsiElement(operator)('<>')
              PsiElement(whitespace)(' ')
          PSValueImpl(Value)
            PSPrefixValueImpl(PrefixValue)
              PSVarImpl(Var)
                PSQualifiedImpl(Qualified)
                  PsiElement(identifier)('y')
                  PsiElement(whitespace)('\n\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSVarImpl(Var)
              PSQualifiedImpl(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSStringLiteralImpl(StringLiteral)
              PsiElement(string)('"Done"')