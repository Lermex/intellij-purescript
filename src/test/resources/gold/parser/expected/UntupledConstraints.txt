Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Prelude')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Console')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Show')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          PsiElement(operator)('<=')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(ClassName)
          PsiElement(proper name)('Nonsense')
          PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PsiElement(identifier)('method')
          PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PSASTWrapperElement(TypeVar)
                  PsiElement(identifier)('a')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Box')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Box')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('showBox')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Show')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(=>)('=>')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ClassName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Show')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('Box')
                  PsiElement(whitespace)(' ')
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('a')
          PsiElement())(')')
          PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('show')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Binder)
            PsiElement(()('(')
            PSASTWrapperElement(Binder)
              PSASTWrapperElement(BinderAtom)
                PSASTWrapperElement(ConstructorBinder)
                  PSASTWrapperElement(Qualified)
                    PsiElement(proper name)('Box')
                    PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Binder)
                    PSASTWrapperElement(VarBinder)
                      PsiElement(identifier)('a')
            PsiElement())(')')
            PsiElement(whitespace)(' ')
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(StringLiteral)
                PsiElement(string)('"Box "')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(IdentInfix)
              PSASTWrapperElement(Qualified)
                PsiElement(operator)('<>')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Value)
              PSASTWrapperElement(PrefixValue)
                PSASTWrapperElement(Var)
                  PSASTWrapperElement(Qualified)
                    PsiElement(identifier)('show')
                    PsiElement(whitespace)(' ')
                PSASTWrapperElement(Var)
                  PSASTWrapperElement(Qualified)
                    PsiElement(identifier)('a')
                    PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('strangeThing')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PsiElement(identifier)('m')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Semigroup')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PsiElement(()('(')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('m')
                          PsiElement(whitespace)(' ')
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeConstructor)
                          PSASTWrapperElement(Qualified)
                            PsiElement(proper name)('Unit')
                    PsiElement())(')')
                    PsiElement(whitespace)(' ')
                  PsiElement(=>)('=>')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('m')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('Unit')
                          PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('m')
                          PsiElement(whitespace)(' ')
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeConstructor)
                          PSASTWrapperElement(Qualified)
                            PsiElement(proper name)('Unit')
                            PsiElement(whitespace)(' ')
                      PsiElement(->)('->')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeVar)
                            PsiElement(identifier)('m')
                            PsiElement(whitespace)(' ')
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeConstructor)
                            PSASTWrapperElement(Qualified)
                              PsiElement(proper name)('Unit')
                              PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('strangeThing')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Binder)
          PSASTWrapperElement(VarBinder)
            PsiElement(identifier)('x')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Binder)
          PSASTWrapperElement(VarBinder)
            PsiElement(identifier)('y')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('x')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('<>')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('y')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(StringLiteral)
              PsiElement(string)('"Done"')